### PR TITLE
fix(workflow): Add back DELETE team member response

### PR DESCRIPTION
**Pull request by scttcper**
We removed the delete response in https://github.com/getsentry/sentry/pull/33443 but we use it in the `/settings/org-id/teams/` page

the leave team function relies on the DELETE response to update the store with the updated team object
https://github.com/getsentry/sentry/blob/8de2382700b365787f47d6091ab689df7c64daea/static/app/actionCreators/teams.tsx#L114-L118

### CODE REVIEW

File: src/sentry/api/endpoints/organization_member_team_details.py [hunk 1]
```diff
@@ -292,4 +292,4 @@ def delete(
             )
             omt.delete()
 
-        return Response(status=200)
+        return Response(serialize(team, request.user, TeamWithProjectsSerializer()), status=200)
```
Comment (AniketDas-Tekky) [inline src/sentry/api/endpoints/organization_member_team_details.py:295]: Do we know if anything relies on the current response format. Will the new body break them?

Comment (scttcper) [reply src/sentry/api/endpoints/organization_member_team_details.py:295]: Right now there is no response

**Review by AniketDas-Tekky — APPROVED** [review]

File: src/sentry/api/endpoints/organization_member_team_details.py (hunk 1 above)
Comment (mdtro) [inline src/sentry/api/endpoints/organization_member_team_details.py:295]: `TeamWithProjectsSerializer()` is tagged as deprecated? Is there another option we could use?

The response also leaks the internal ID and would get flagged as an IDOR vulnerability. Is there specific subset of the data the frontend needs that can be returned?

https://github.com/getsentry/sentry/blob/48c55acdbf09b408160e0386483380b72cb0aec9/src/sentry/api/serializers/models/team.py#L254-L258

Comment (scttcper) [reply src/sentry/api/endpoints/organization_member_team_details.py]: how's the TeamSerializer sound? We do not need the projects.

**Review by mdtro — APPROVED** [review]

**Pull request merged**